crates/wasm/src/tx.rs

```diff
@@ -172,11 +173,54 @@ pub fn build(
 
     let plan: TransactionPlan = plan_proto.try_into()?;
 
-    let tx: Transaction = plan
-        .build(&fvk, witness_data_proto.try_into()?)?
-        .authorize(&mut OsRng, &auth_data_proto.try_into()?)?;
+    let tx: Transaction = plan.build(
+        &fvk,
+        &witness_data_proto.try_into()?,
+        &auth_data_proto.try_into()?,
+    )?;
 
     let value = serde_wasm_bindgen::to_value(&tx.to_proto())?;
+
+    Ok(value)
+}
+
+/// Build parallel tx
+/// Building a transaction may take some time,
+/// depending on CPU performance and number of actions in transaction_plan
+/// Arguments:
+///     actions: `Vec<Actions>`
+///     transaction_plan: `pb::TransactionPlan`
+///     witness_data: `pb::WitnessData`
+///     auth_data: `pb::AuthorizationData`
+/// Returns: `pb::Transaction`
+#[wasm_bindgen]
+pub fn build_parallel(
```

Comment:
I can't really tell, but is this where web workers will be spawned later?

Reply:
The web-workers will call the `build_action` method in the wasm planner, and then the prebuilt actions will be slotted into `build_parallel` where the final transaction can be assembled.